### PR TITLE
refactor: lock wallet on login timeout

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

File: src/composables/auth.ts [hunk 1]
```diff
@@ -69,6 +70,7 @@ export const useAuth = createCustomScopedComposable(() => {
   const isAuthenticating = ref(false);
   const isMnemonicRestored = ref(false);
   const encryptionKey = ref<CryptoKey>();
+  const isSessionExpired = ref(false);
```
Comment (peronczyk) [inline src/composables/auth.ts:73]: This can be a regular `let` as it doesn't affect any UI element.

File: src/composables/auth.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ import { useModals } from './modals';
 import { useStorageRef } from './storageRef';
 
 const CHECK_FOR_SESSION_KEY_INTERVAL = 5000;
+let expirationTimeout: NodeJS.Timeout;
```
Comment (peronczyk) [inline src/composables/auth.ts:47]: This can be placed inside of the composable as it is using the `createCustomScopedComposable`.

**Review by peronczyk — APPROVED** [review]

**Comment by Liubov-crypto** [comment]
Timeout lock isn't working for extension.

**Comment by Liubov-crypto** [comment]
Timeout lock isn't working for extension when I'm setting 1 or 5 min timer

**Comment by Liubov-crypto** [comment]
LGTM

File: src/background/bgPopupHandler.ts [hunk 3]
```diff
@@ -88,13 +88,26 @@ export const removePopup = (id: string) => delete popups[id];
...
+let keepAliveInterval: NodeJS.Timeout;
 
 export const setSessionTimeout = async (sessionTimeout: number) => {
   if (expirationTimeout) {
     clearTimeout(expirationTimeout);
   }
+  if (keepAliveInterval) {
+    clearInterval(keepAliveInterval);
+  }
+  /**
```
Comment (peronczyk) [inline src/background/bgPopupHandler.ts:100]: Please add empty line before this one.

File: src/composables/auth.ts [hunk 4]
```diff
@@ -365,23 +368,39 @@ export const useAuth = createCustomScopedComposable(() => {
...
   }
 
+  watch(
+    isAuthenticated,
+    (val) => {
+      if (val) {
+        isSessionExpired = false;
+      }
+    },
+  );
```
Comment (peronczyk) [inline src/composables/auth.ts:378]: I suggest to add some comment why we need this and how this affects the app.

**Pull request merged**